infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -102,20 +92,29 @@ private boolean processMsg(LogReplicationEntryMsg txMessage) {
 
             for (OpaqueEntry opaqueEntry : newOpaqueEntryList) {
                 for (UUID streamId : opaqueEntry.getEntries().keySet()) {
-                    if (!streamMap.containsKey(streamId)) {
+                    if (ignoreEntriesForStream(streamId)) {
                         log.warn("Skip applying log entries for stream {} as it is noisy. LR could be undergoing a rolling upgrade", streamId);
                         continue;
                     }
 
                     List<SMREntry> smrEntries = opaqueEntry.getEntries().get(streamId);
                     if (streamId.equals(REGISTRY_TABLE_ID)) {
                         // Only retain tables that are not present in the registry table
-                        smrEntries = fetchNewEntries(new ArrayList<>(smrEntries));
-                    }
-
-                    for (SMREntry smrEntry : smrEntries) {
-                        // If stream tags exist for the current stream, it means its intended for streaming on the Sink (receiver)
-                        txnContext.logUpdate(streamId, smrEntry, dataStreamToTagsMap.get(streamId));
+                        List<SMREntry> newEntries = fetchNewEntries(new ArrayList<>(smrEntries));
+                        for (SMREntry smrEntry : newEntries) {
+                            // If stream tags exist for the current stream, it means its intended for streaming on the Sink (receiver)
+                            txnContext.logUpdate(streamId, smrEntry, config.getDataStreamToTagsMap().get(streamId));
```

Comment:
nit: L106 and 116 are common for both cases and can be moved outside if.. else

Reply:
L106 was for applying RegistryTable entries, after which a re-sync is triggered, that's the reason they are put into a separate clause